test/app-luatest/uri_unit_test.lua

```diff
@@ -0,0 +1,68 @@
+local build_path = os.getenv("BUILDDIR")
+package.cpath = build_path .. '/src/lua/?.lua;' .. package.cpath
+local uri = require('src.lua.uri')
+local t = require('luatest')
+
+local decimal = require('decimal')
+local datetime = require('datetime')
+
+local g = t.group('uri_unit')
+
+local uri_params_g = t.group("uri_params", {
+    { params = { }, query_string = "" },
+    { params = nil, query_string = "" },
+    { params = { key = {} }, query_string = "key" },
+    { params = { key = "" }, query_string = "key=" },
+    { params = { db1 = "tarantool",  db2 = "redis" }, query_string = "db1=tarantool&db2=redis" },
+    { params = { key = "tarantool" }, query_string = "key=tarantool" },
+    { params = { boolean_type = false }, query_string = "boolean_type=false" },
+    { params = { boolean_type = true }, query_string = "boolean_type=true" },
+    { params = { integer_type = 50 }, query_string = "integer_type=50" },
+    { params = { decimal_type = decimal.new(10) }, query_string = "decimal_type=10" },
+    { params = { datetime_type = datetime.new() }, query_string = "datetime_type=1970-01-01T00:00:00Z" },
+    { params = { int64_type = tonumber64(10) }, query_string = "int64_type=10" },
```

Comment:
Minor: I guess, it's better to use `-1ULL` here to encode the real 64-bit integer (consider #4738).